torch/nn/modules/module.py

```diff
@@ -76,6 +76,9 @@ def register_buffer(self, name, tensor):
         Example:
             >>> self.register_buffer('running_mean', torch.zeros(num_features))
         """
+        if hasattr(self, name) and name not in self._buffers.keys():
```

Comment:
Do you need the `.keys()` here? Can't you just do instead `and name not in self._buffers`? Same for the other occurences.

Reply:
Didn't know you could even do that 👍 - will change that and push